### PR TITLE
Set Up Traveler Class

### PR DESCRIPTION
**Pull request by BrigetteDoelp**
#### What’s this PR do?
- ​Sets up the `Traveler` class.
- Begins testing for `Traveler` class.
- Finds trips for one traveler.
- Imports `moment js`.
#### Where should the reviewer start?
​In `index.js`, moving into `Traveler.js`, then into `Traveler-test.js`.
#### How should this be manually tested?
​By running `npm test`, and opening the site through the terminal, run `npm start` and go to `localhost:8080`. Open the dev tools and make sure the appropriate data is being logged.
#### Any background context you want to provide?
​This is incomplete, and there will be data missing. 
#### What are the relevant tickets?
​Revolves #4 
​Revolves #5 
​Revolves #2 
​Revolves #6

### CODE REVIEW

File: src/Traveler.js [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
-  constructor(id, name, travelerType) {
-    this.id = id;
-    this.name = name;
-    this.travelerType = travelerType;
+  constructor(traveler, trips) {
+    this.id = traveler.id;
+    this.name = traveler.name;
+    this.travelerType = traveler.travelerType;
+    this.allTrips = trips;
+    this.travelersTrips = [];
```
Comment (holladayian) [inline src/Traveler.js:11]: hmm, I'm still not sure that traveler needs access to everybody else's trips.

File: src/Traveler.js [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
+  }
+
+  findTrips() {
+    this.travelersTrips = this.allTrips.filter(trip => trip.userID === this.id)
+  };
```
Comment (holladayian) [inline src/Traveler.js:16]: I think if you wanted to refactor, you could separate this out into a some kind of TripRepository. That being said, the word "Repository" makes me physically ill.

File: src/Traveler.js [hunk 3]
```diff
@@ -1,9 +1,28 @@
...
+
+  pastTrips() {
+    let today = moment(Date.now())
```
Comment (holladayian) [inline src/Traveler.js]: After (minimal) research, I think moment() will return today. Not a hundo on that though. Food for thought.

**Review by holladayian — APPROVED** [review]
Functionality looks sweet!! And yeah, I'd recommend having a user only able to access their own trips. There's probably a bajillion different ways to do this though.

File: webpack.config.js [hunk 4]
```diff
@@ -36,6 +37,8 @@ module.exports = {
     new HtmlWebpackPlugin({
       template: './src/index.html'
     })
+    new MomentLocalesPlugin({
+            localesToKeep: ['es-us'],
```
Comment (holladayian) [inline webpack.config.js]: hmm

**Review by holladayian — COMMENTED** [review]
I still don't even know about this. Mine doesn't have it. But then again mine will probably explode, so who know!

File: test/Traveler-test.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    console.log(traveler)
+  })
+
+  // it('is a function', () =>{
+  //   expect(Traveler).to.be.a('function');
+  // });
+  //
+  // it('should be an instance of Traveler', () => {
+  //   expect(traveler).to.be.an.instanceof(Traveler);
+  // });
```
Comment (holladayian) [inline test/Traveler-test.js:28]: ?? Did you want to comment these back in for testing porpoises?
porpoise

**Review by holladayian — COMMENTED** [review]
Unsure if these were commented back in. Just checkin!

File: src/index.js [hunk 6]
```diff
@@ -11,34 +11,47 @@ import './images/turing-logo.png'
...
 window.onload = onLoadContent;
+// window.onload = fuck;
+
 
 function onLoadContent() {
-  let promise1 = api.fetchAllTravelers();
-  let promise2 = api.fetchAllTrips();
-  let promise3 = api.fetchAllDestinations();
-  console.log(moment().format('MMMM Do YYYY'));
+  let userID = Math.floor((Math.random() * 50) + 1);
```
Comment (holladayian) [inline src/index.js:28]: Noice! I don't think we'll need this eventually though :( For it3 a user will log in with their credentials and that will give us the information to fetchOneTraveler with

File: src/index.js [hunk 7]
```diff
@@ -11,34 +11,47 @@ import './images/turing-logo.png'
...
-      trips = values[1].trips;
+      traveler = values[2]
+      trips = values[0].trips;
       destinations = values[2].destinations;
+      frick()
     })
 }
 
 //GENERAL FUNCTIONS
+function frick(solo, trippys) {
```
Comment (holladayian) [inline src/index.js]: Hilarious! But probably should rename for professionally. What am I, the professionally police? You do what you want!

**Review by holladayian — COMMENTED** [review]
Functionality looks solid! You really understand Promises

**Comment by holladayian** [comment]
Nice work overall! You do have some double commits here and there. Maybe more concise descriptions of what each is committing might help there, but overall good work!

File: src/Traveler.js (hunk 1 above)
Comment (BrigetteDoelp) [reply src/Traveler.js:11]: Yeah, I need to go back and refactor that for sure.

File: src/Traveler.js (hunk 2 above)
Comment (BrigetteDoelp) [reply src/Traveler.js:16]: LOL

File: webpack.config.js (hunk 4 above)
Comment (BrigetteDoelp) [reply webpack.config.js]: I don't know how that got there its gone now

File: test/Traveler-test.js (hunk 5 above)
Comment (BrigetteDoelp) [reply test/Traveler-test.js:28]: Yep! I did after I initially pushed this. haahah Thank you

File: src/index.js (hunk 6 above)
Comment (BrigetteDoelp) [reply src/index.js:28]: Yea, I mostly have this bit in for testing purposes. Every time the page loads it boots up with a new user and let's me know that everything is dynamicc

File: src/index.js (hunk 7 above)
Comment (BrigetteDoelp) [reply src/index.js]: I. Don't remember naming that.

**Pull request merged**